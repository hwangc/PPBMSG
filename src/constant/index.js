export const SUCCESS_FETCH_ORDER = 'SUCCESS_FETCH_ORDER';
export const FAIL_FETCH_ORDER = 'FAIL_FETCH_ORDER';
export const START_FETCH_ORDER = 'START_FETCH_ORDER';
export const SUCCESS_FETCH_LOC = 'SUCCESS_FETCH_LOC';
export const FAIL_FETCH_LOC = 'FAIL_FETCH_LOC';
export const START_FETCH_LOC = 'START_FETCH_LOC';
export const types = {
  START_FETCH_ORDER,
  SUCCESS_FETCH_ORDER,
  FAIL_FETCH_ORDER,

  START_FETCH_LOC,
  SUCCESS_FETCH_LOC,
  FAIL_FETCH_LOC
};
export const CHUU = 'chuu';
export const MOSSBEAN = 'mossbean';
export const ICECREAM12 = 'icecream12';
export const MALL = {
  CHUU,
  MOSSBEAN,
  ICECREAM12
};
export const initialStateOrder = {
  id: null,
  prdNum: null,
  orderAddress: null,
  itemsInfo: [],
  error: null
};

export const initialStateMap = {
  latitude: 0,
  longitude: 0,
  latitudeDelta: 0.0922,
  longitudeDelta: 0.0421
};

export const ORDER = {
  order_no: 'o_1',
  product: [
    { product_no: 'p_1' },
    { product_no: 'p_2' },
    { product_no: 'p_3' }
  ],
  ship_address1: 'hello',
  ship_address2: 'address'
};
const GOOGLE = 'google';
const DAUM = 'daum';
const FREEGEOIP = 'freegeoip';
export const GEO_API = {
  GOOGLE,
  DAUM,
  FREEGEOIP
};
