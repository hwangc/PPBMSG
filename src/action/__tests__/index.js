import * as actions from '../index';
import { CHUU, ORDER } from '../../constant';
import mockStore from 'redux-mock-store';

const store = mockStore({});
global.fetch = jest.fn().mockImplementation(url => {
  if (url) {
    return Promise.resolve({
      json: () => ({
        response: {
          result: ORDER
        }
      })
    });
  } else {
    return null;
  }
});

describe('PPBMSG Action Creators => ', () => {
  beforeEach(() => {
    store.clearActions();
  });

  describe('SYNC ', () => {
    test('successFetchOrderAction => should create an order payload', () => {
      expect(actions.successFetchOrderAction(ORDER)).toMatchSnapshot();
    });
  });

  describe('ASYNC ', () => {
    test('fetchOrder(CHUU) => should return success action', async () => {
      await store.dispatch(actions.fetchOrder(CHUU));
      expect(store.getActions()).toMatchSnapshot();
    });
    test('fetchOrder(NULL) => should return fail action', async () => {
      await store.dispatch(actions.fetchOrder(null));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
