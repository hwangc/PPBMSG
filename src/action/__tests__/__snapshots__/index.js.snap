// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PPBMSG Action Creators =>  ASYNC  fetchOrder(CHUU) => should return success action 1`] = `
Array [
  Object {
    "payload": Object {
      "error": null,
      "id": null,
      "itemsInfo": Array [],
      "orderAddress": null,
      "prdNum": null,
    },
    "type": "START_FETCH_ORDER",
  },
  Object {
    "payload": Object {
      "error": "",
      "id": "o_1",
      "itemsInfo": Array [
        Object {
          "image": undefined,
          "name": undefined,
          "price": undefined,
        },
      ],
      "orderAddress": "hello address",
      "prdNum": 3,
    },
    "type": "SUCCESS_FETCH_ORDER",
  },
  Object {
    "payload": Object {
      "location": Object {
        "latitude": 37.551954,
        "latitudeDelta": 0.0922,
        "longitude": 127.058964,
        "longitudeDelta": 0.0421,
      },
    },
    "type": "SUCCESS_FETCH_LOC",
  },
]
`;

exports[`PPBMSG Action Creators =>  ASYNC  fetchOrder(NULL) => should return fail action 1`] = `
Array [
  Object {
    "payload": Object {
      "error": null,
      "id": null,
      "itemsInfo": Array [],
      "orderAddress": null,
      "prdNum": null,
    },
    "type": "START_FETCH_ORDER",
  },
  Object {
    "payload": Object {
      "error": "",
      "id": "o_1",
      "itemsInfo": Array [
        Object {
          "image": undefined,
          "name": undefined,
          "price": undefined,
        },
      ],
      "orderAddress": "hello address",
      "prdNum": 3,
    },
    "type": "SUCCESS_FETCH_ORDER",
  },
  Object {
    "payload": Object {
      "location": Object {
        "latitude": 37.551954,
        "latitudeDelta": 0.0922,
        "longitude": 127.058964,
        "longitudeDelta": 0.0421,
      },
    },
    "type": "SUCCESS_FETCH_LOC",
  },
]
`;

exports[`PPBMSG Action Creators =>  SYNC  successFetchOrderAction => should create an order payload 1`] = `
Object {
  "payload": Object {
    "error": "",
    "id": undefined,
    "itemsInfo": undefined,
    "orderAddress": "undefined undefined",
    "prdNum": 0,
  },
  "type": "SUCCESS_FETCH_ORDER",
}
`;
